nifi-api/pom.xml

```diff
@@ -22,5 +22,13 @@
     </parent>
     <artifactId>nifi-api</artifactId>
     <packaging>jar</packaging>
-    <!-- This module should kept to having no dependencies -->
+
+    <!-- This module should kept to having no dependencies except nifi-versioned-components -->
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-versioned-components</artifactId>
+            <version>1.14.0-SNAPSHOT</version>
+        </dependency>
+    </dependencies>
```

Comment:
Is there a specific reason for introducing `nifi-versioned-components` as a dependency to `nifi-api` in this pull request?  There do not appear to be any other changes to `nifi-api` in this pull request, so it seems like this change is not necessary right now.  In addition, this introduces the transitive dependency on `swagger-annotations`.